client/src/main/java/org/dst/client/DstDictProxy.java

```
@@ -0,0 +1,147 @@
+package org.dst.client;
+
+import org.dst.exception.DictKeyNotFoundException;
+import org.dst.exception.DstException;
+import org.dst.exception.KeyNotFoundException;
+import org.dst.server.generated.CommonProtocol;
+import org.dst.server.generated.DictProtocol;
+import org.dst.server.service.DstDictService;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DstDictProxy {
+
+  private DstDictService service;
+
+  public DstDictProxy(DstDictService service) {
+    this.service = service;
+  }
+
+  // Put a new dict
+  public void put(String key, Map<String, String> dict) {
+    DictProtocol.PutRequest.Builder request =
+        DictProtocol.PutRequest.newBuilder();
+    request.setKey(key);
+    DictProtocol.DstDict dstDict = DictUtil.buildDstDict(dict);
+    request.setDict(dstDict);
+    DictProtocol.PutResponse response = service.put(request.build());
+    if (response.getStatus() != CommonProtocol.Status.OK) {
+      throw new DstException(String.format("Error code is %d",
+          response.getStatus().getNumber()));
+    }
+  }
+
+  // Get a dict
+  public Map<String, String> get(String key) {
+    Map<String, String> dict = new HashMap();
+    DictProtocol.GetRequest.Builder request =
+        DictProtocol.GetRequest.newBuilder();
+    request.setKey(key);
+    DictProtocol.GetResponse response =
+        service.get(request.build());
+    if (response.getStatus() != CommonProtocol.Status.OK) {
+      throw new KeyNotFoundException(key);
```

We should throw key not found exception only if status equals KEY_NOT_FOUND.